File: src/main/scala/baseSpark/createDataFrameDemo/demo1.scala
package baseSpark.createDataFrameDemo

import org.apache.spark.sql.{SQLContext, SaveMode}
import org.apache.spark.{SparkConf, SparkContext}

/**
  * Created by wanglei on 2018/1/2.
  * 通过json文件创建dataFrame
  */
object demo1 {
  def main(args: Array[String]): Unit = {
    val hdfsParquetFileDir = "hdfs://master-yyft:9000/tmp/hive/test/"
    val localParquetFileDir = "/home/bigdata/IdeaProjects/MySpark/src/testData/ParquetDir/"

    val sparkConf = new SparkConf().setAppName(getClass.getName.replace("$", "")).setMaster("local[1]")
    val sc = new SparkContext(sparkConf)
    val sqlContext = new SQLContext(sc)

    //读取json文件的方式生成dataFrame
    val df = sqlContext.read.json("/home/bigdata/IdeaProjects/MySpark/src/testData/jsonData.txt")
    df.show()


    //dataFrame的write默认保存的是parquet格式
    df.write.mode(SaveMode.Overwrite).save(hdfsParquetFileDir)

    //读取parquet文件的方式生成dataFrame
    val df2 = sqlContext.read.parquet(hdfsParquetFileDir)
    df2.show()

    sqlContext.


  }

}

File: src/main/java/base/ioDemo/ReaderWriterDemo/demo1.java
package base.ioDemo.ReaderWriterDemo;

import java.io.FileReader;
import java.io.FileWriter;
import java.io.IOException;

/**
 * Created by wanglei on 2018/1/17.
 * 文件拷贝
 * <p>
 * 一个字符等于2个字节
 * 字符流不能用于拷贝文件
 */
public class demo1 {
    public static void main(String[] args) throws IOException {
        FileReader fr = new FileReader("F:\\IdeaProjects\\MySpark\\src\\testData\\fr.txt");
        FileWriter fw = new FileWriter("F:\\IdeaProjects\\MySpark\\src\\testData\\fw.txt");
        int i;
        while ((i = fr.read()) != -1) {
            fw.write(i);
        }
        fr.close();
        fw.close(); //Writer类中有一个2k的小缓冲区，如果不关流，就会将内容写到缓冲区里。关流，会将缓冲区内容刷新写出，再关闭。
    }
}
